Test/Unit/Model/DutyCalculatorTest.php

```diff
@@ -0,0 +1,279 @@
+
+    /**
+     * @var StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var PriceCurrencyInterface
+     */
+    protected $priceCurrency;
+
+    /**
+     * @var \Reach\Payment\Model\ResourceModel\CsvHsCodeFactory
+     */
+    protected $csvHsCodeFactory;
+
+    /**
+     * @var \Magento\Directory\Model\Region
+     */
+    protected $regionModel;
+
+    /**
+     * @var \Reach\Payment\Model\Api\HttpRestFactory
+     */
+    protected $httpRestFactory;
+
+    /**
+     * @var \Reach\Payment\Api\Data\DutyResponseInterface
+     */
+    protected  $response;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+     protected $logger;
+
+    protected function setUp(): void {
+
+        $this->reachHelper = $this->createMock('\Reach\Payment\Helper\Data');
+        $this->quoteRepository = $this->createMock('\Magento\Quote\Api\CartRepositoryInterface');
+        $this->quoteIdMaskFactory = $this->createMock('\Magento\Quote\Model\QuoteIdMaskFactory');
+        $this->scopeConfig = $this->createMock('ScopeConfigInterface');
+        $this->storeManager = $this->createMock('StoreManagerInterface');
+        $this->priceCurrency = $this->createMock('PriceCurrencyInterface');
+        $this->csvHsCodeFactory = $this->createMock('\Reach\Payment\Model\ResourceModel\CsvHsCodeFactory');
+        $this->regionModel = $this->createMock('\Magento\Directory\Model\Region');
+        $this->response = $this->createMock('\Reach\Payment\Api\Data\DutyResponseInterface');
+        $this->logger = $this->createMock('\Psr\Log\LoggerInterface');
+
+        $this->httpRestFactory = $this->getMockBuilder('Reach\Payment\Model\Api\HttpRestFactory')
+            ->disableOriginalConstructor()->setMethods(['create'])
+            ->getMock();
+
+        $this->checkoutSession = $this->getMockBuilder('\Magento\Checkout\Model\Session')
+            ->disableOriginalConstructor()
+            ->getMock();
+    }
+
+    protected function tearDown(): void {
+        unset($this->reachHelper);
+        unset($this->quoteRepository);
+        unset($this->quoteIdMaskFactory);
+        unset($this->scopeConfig);
+        unset($this->storeManager);
+        unset($this->priceCurrency);
+        unset($this->csvHsCodeFactory);
+        unset($this->regionModel);
+        unset($this->response);
+        unset($this->httpRestFactory);
+        unset($this->checkoutSession);
+        unset($this->logger);
+    }
+
+    public function testGetDutyTaxAPISuccess() {
+
+        $testSeam = new DutyCalculatorSeam(
+            $this->reachHelper,
+            $this->checkoutSession,
+            $this->quoteRepository,
+            $this->quoteIdMaskFactory,
+            $this->scopeConfig,
+            $this->storeManager,
+            $this->priceCurrency,
+            $this->csvHsCodeFactory,
+            $this->regionModel,
+            $this->httpRestFactory,
+            $this->response,
+            $this->logger);
+
+        $json = '{
```

Comment:
could put this in seperate file to shorten up the test

Reply:
Will be resolved in this Sprint.